### PR TITLE
[WIP] Misc. Documentation fixes

### PR DESCRIPTION
**Pull request by renemilk**
I'll be fixing a couple of missing parameters in docstrings and such. Maybe also unexpected ones like
https://github.com/pymor/pymor/commit/893f5634f6349e96855b4bf967f2cca5190976c3. I'm not sure however that's how one should deal with the situation. Asking for input my fellow @pymor/pymor-devs

### CODE REVIEW

File: src/pymor/parameters/base.py [hunk 1]
```diff
@@ -380,7 +380,7 @@ def strip_parameter(self, mu):
         assert all(getattr(mu.get(k, None), 'shape', None) == v for k, v in self.parameter_type.items())
         return Parameter({k: mu[k] for k in self.parameter_type})
 
-    def build_parameter_type(self, *args, **kwargs):
+    def build_parameter_type(self, *args, provides=None, **kwargs):
```
Comment (sdrave) [inline src/pymor/parameters/base.py]: This is invalid syntax in Python 2 and I would be surprised if 3to2 can handle that ...

Comment (sdrave) [inline src/pymor/parameters/base.py]: This is invalid syntax in Python 2 and I would be surprised if 3to2 can handle that ...

**Comment by sdrave** [comment]
> I'm not sure however that's how one should deal with the situation.

What do you mean by "the situation"?

File: src/pymor/parameters/base.py (hunk 1 above)
Comment (renemilk) [inline src/pymor/parameters/base.py]: I've checked now and 3to2 transforms this correctly, albeit into somewhat ugly code.

File: src/pymor/core/defaults.py [hunk 2]
```diff
@@ -274,9 +274,7 @@ def defaults(*args, **kwargs):
     # FIXME this will have to be adapted for Python 3
 
     assert all(isinstance(arg, str) for arg in args)
-    assert set(kwargs.keys()) <= {'sid_ignore', 'qualname'}
-    sid_ignore = kwargs.get('sid_ignore', ())
-    qualname = kwargs.get('qualname', None)
+    assert set(kwargs.keys()) != set()
```
Comment (sdrave) [inline src/pymor/core/defaults.py]: No, kwargs would actually have to be empty (-> simply remove kwargs)

File: src/pymor/core/defaults.py [hunk 3]
```diff
@@ -274,9 +274,7 @@ def defaults(*args, **kwargs):
     # FIXME this will have to be adapted for Python 3
```
Comment (sdrave) [inline src/pymor/core/defaults.py]: I guess we can remove this line ...

File: src/pymor/tools/mpi.py [hunk 4]
```diff
@@ -284,7 +284,7 @@ def method_call(obj_id, name_, *args, **kwargs):
     obj_id
         The :class:`ObjectId` of the object on which to call
         the method.
-    `name_`
+    name_
```
Comment (sdrave) [inline src/pymor/tools/mpi.py]: If I remember correctly, the backticks were needed to make sphinx happy ...

Comment (renemilk) [inline src/pymor/tools/mpi.py]: Yep, you're right.

**Pull request merged**